packages/typescript-estree/src/convert.ts

```diff
@@ -1917,7 +1917,13 @@ export class Converter {
         return result;
       }
 
-      case SyntaxKind.TaggedTemplateExpression:
+      case SyntaxKind.TaggedTemplateExpression: {
+        if (node.tag.flags & ts.NodeFlags.OptionalChain) {
+          this.#throwError(
+            node,
+            'Cannot have an optional chain in TaggedTemplateExpression.',
```

Comment:
Final change -- let's match TS's error message
```suggestion
            'Tagged template expressions are not permitted in an optional chain.',
```

you'll need to regen the snapshots to match the error text